_pytest/terminal.py

```diff
@@ -152,9 +152,9 @@ def __init__(self, config, file=None):
         self.reportchars = getreportopt(config)
         self.hasmarkup = self._tw.hasmarkup
         self.isatty = file.isatty()
-        self._progress_items_reported = 0
-        self._show_progress_info = (self.config.getoption('capture') != 'no' and
-                                    self.config.getini('console_output_style') == 'progress')
+        self._progress_nodeids_reported = set()
+        self._show_progress_info = (self.config.getoption('capture') != 'no' and not self.config.getoption('setupshow')
```

Comment:
that condition is starting to look complex enough for putting it in a small function

Reply:
Good idea, thanks. Done.